barretenberg/cpp/src/barretenberg/honk/sumcheck/sumcheck_round.hpp

```diff
@@ -425,6 +425,8 @@ template <typename Flavor> class SumcheckVerifierRound {
         // with a simulated builder.
         bool sumcheck_round_failed(false);
         if constexpr (IsRecursiveFlavor<Flavor>) {
+            // TODO(#726): Need to constrain this equality and update the native optional return value mechanism for the
```

Comment:
Should there be an optional return at all? If the recursive proof fails, then the native proof should fail, too, right?

Reply:
Right, just need to figure out how to resolve re: previous comment.